mongodbatlas/resource_mongodbatlas_global_cluster_config.go

```diff
@@ -348,3 +379,36 @@ func addManagedNamespaces(ctx context.Context, conn *matlas.Client, add []interf
 
 	return nil
 }
+
+func expandCustomZoneMapping(tfMap map[string]interface{}) *matlas.CustomZoneMapping {
+	if tfMap == nil {
+		return nil
+	}
+
+	apiObject := &matlas.CustomZoneMapping{
+		Location: tfMap["location"].(string),
+		Zone:     tfMap["zone"].(string),
+	}
+
+	return apiObject
+}
+
+func expandCustomZoneMappings(tfList []interface{}) []matlas.CustomZoneMapping {
+	if len(tfList) > 0 {
+		var apiObjects []matlas.CustomZoneMapping
+
+		for _, tfMapRaw := range tfList {
+			if tfMap, ok := tfMapRaw.(map[string]interface{}); ok {
+				apiObject := expandCustomZoneMapping(tfMap)
+				if apiObject == nil {
+					continue
+				}
+				apiObjects = append(apiObjects, *apiObject)
+			}
+		}
+
+		return apiObjects
+	}
+
+	return nil
```

Comment:
[nit] couple of things, prefer an early exit like with `expandCustomZoneMapping`, 
```go
if len(tfList) == 0 {
  return nil
}
```

second since you know the length of tfList (`len(tfList)) you can optimise memory allocation  of `apiObjects` by initializing it to the right length and also avoid `append` by directly using indexes

Reply:
understood, will make changes